numpy/f2py/tests/test_callback.py

```diff
@@ -59,12 +59,11 @@ class TestF77Callback(util.F2PyTest):
        end
     """
 
-    @pytest.mark.slow
     @pytest.mark.parametrize('name', 't,t2'.split(','))
     def test_all(self, name):
         self.check_function(name)
 
-    @pytest.mark.slow
+    @pytest.mark.xfail(IS_PYPY, reason="PyPy does not modify tp_doc")
```

Comment:
What do you mean by this? Is this a numpy bug caused by setting `tp_doc` too late?

Reply:
yes, like in `numpy.core._add_newdocs which uses `add_docstring` from `_multiarrayumath`.

Reply:
Would this be more correct?
```suggestion
    @pytest.mark.xfail(IS_PYPY, reason="tp_doc is set after PyType_Ready is called, which pyp does not support")
```

Reply:
changing to "PyPy cannot modify tp_doc after PyType_Ready"